### PR TITLE
Add docs for connecting to Databricks and Spark in dbt Cloud

### PR DESCRIPTION
**Pull request by drewbanin**
## Description & motivation
This PR is intended to document the required and optional fields available when configuring a new Databricks or Spark connection in dbt Cloud.

I took some... liberties... with the screenshots for clarity and will seek to make sure that the product reflects helpful descriptions commensurate with what's shown in these docs :)

## To-do before merge
@jtcohen6 could I ask you to give this a once-over and tell me if anything I said in here doesn't feel exactly right?

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

### CODE REVIEW

File: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md [hunk 1]
```diff
@@ -142,3 +142,50 @@ more information on the initial configuration of a BigQuery OAuth connection in
...
+<Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/databricks-connection.png" title="Configuring a Databricks connection"/>
+
+## Connecting to Apache Spark
+
+### HTTP and Thrift
+
+dbt Cloud supports connecting to an Apache Spark cluster using the HTTP method
+or the Thrift method.
+
+The following fields are available when creating an Apache Spark connection over HTTP:
```
Comment (jtcohen6) [inline website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md]: Did you mean to include both Thrift + HTTP methods below, or just HTTP?

File: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md [hunk 2]
```diff
@@ -142,3 +142,50 @@ more information on the initial configuration of a BigQuery OAuth connection in
...
+| ----- | ----------- | -------- |
+| Host Name | The hostname of the Spark cluster to connect to | `yourorg.sparkhost.com` |
+| Port | The port to connect to Spark on | 443 |
+| Organization | Optional, only required for Azure Databricks | 0123456789 |
+| Cluster | The ID of the cluster to connect to | 1234-567890-abc12345 |
+| Connection Timeout | Number of seconds after which to timeout a connection | 10 |
+| Connection Retries | Number of times to attempt connecting to cluster before failing | 10 |
+| User | Optional | dbt_cloud_user |
+| Auth | Optional, supply if using Kerberos | `KERBEROS` |
+| Kerberos Service Name | Optional, supply if using Kerberos | `hive` |
```
Comment (jtcohen6) [inline website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md:193]: These are only for the `Thrift` method, not `HTTP`. There's a full table breaking down what's for what in the [dbt-spark readme](https://github.com/fishtown-analytics/dbt-spark#configuring-your-profile)

File: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md [hunk 3]
```diff
@@ -142,3 +142,50 @@ more information on the initial configuration of a BigQuery OAuth connection in
...
+| User | Optional | dbt_cloud_user |
+
+<Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/databricks-connection.png" title="Configuring a Databricks connection"/>
+
+## Connecting to Apache Spark
+
+### HTTP and Thrift
+
+dbt Cloud supports connecting to an Apache Spark cluster using the HTTP method
+or the Thrift method.
```
Comment (jtcohen6) [inline website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md]: The inclusion of Databricks-relevant properties below (e.g. `Organization`) may be confusing, so you might want to include a note to the effect of:

```suggestion
dbt Cloud supports connecting to an Apache Spark cluster using the HTTP method
or the Thrift method. Note that, while the HTTP method can be used to connect to an all-purpose Databricks cluster, the ODBC method is recommended for all Databricks connections.
```

**Review by jtcohen6 — APPROVED** [review]
Left a few comments, feel free to handle or dismiss. Assuming you do, lgtm!

File: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md (hunk 2 above)
Comment (drewbanin) [reply website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md:193]: thanks jerco! The form we use in dbt Cloud could definitely make this easier - I am just going to link out to the readme for now, but we can circle back and improve the connection creation form + docs at some point in the future. I appreciate the pointer!

**Review by drewbanin — COMMENTED** [review]
Ok - all feedback addressed in one way or another - merging when the build passes!

**Pull request merged**